pandas/plotting/_matplotlib/core.py

```diff
@@ -120,7 +121,77 @@ def __init__(
 
         self.sort_columns = sort_columns
 
-        self.subplots = subplots
+        if isinstance(subplots, bool):
+            self.subplots = (
+                subplots
+            )  # type: Optional[Union[bool, Sequence[Sequence[Union[str, int]]]]]
```

Comment:
for some reason I have to declare the type here once again despite having specified it already above.

Also note I have to add `int` so that mypy doesn't complain, but users should only pass strings (column names) instead of ints (indices)

Reply:
ci failing with Check for use of comment-based annotation syntax. please use py3.6+ syntax https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html?highlight=variable%20annotation#explicit-types-for-variables

> for some reason I have to declare the type here once again despite having specified it already above.

from https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#type-inference

> Mypy considers the initial assignment as the definition of a variable.

self.subplots is hence infered to be bool without the variable type annotation. (and the mypy error then makes sense `error: Incompatible types in assignment (expression has type "List[<nothing>]", variable has type "bool")`

could use an alias to reduce duplication, and/or could declare variable type in class scope to reduce noise.

also maybe Sequence[Sequence[Union[str, int]]]]] -> Sequence[Sequence[Label]]]]

Reply:
I think that this is adding quite a bit of code to \_\_init__. maybe create a _validate_subplot_kwarg method (the variable annotation issue could disappear (with function parameters and return type typed)